features/networking/ovn-egress-ip.feature

```diff
@@ -643,3 +643,194 @@ Feature: OVN Egress IP related features
+  @4.9
+  Scenario: lr-policy-list and snat should be updated correctly after remove egressip objects 
+    Given I store the schedulable workers in the :nodes clipboard
+    Then label "k8s.ovn.org/egress-assignable=true" is added to the "<%= cb.nodes[0].name %>" node
+
+    #Get unused IP as egress ip
+    Given I store a random unused IP address from the reserved range to the clipboard
+
+    #Create first project and pods in it,add label to the namespace
+    Given I have a project
+    And evaluation of `project.name` is stored in the :proj1 clipboard
+    When I run the :label admin command with:
+      | resource | namespace           |
+      | name     | <%= cb.proj1 %>     |
+      | key_val  | org=pm              |
+    Then the step should succeed
+
+    #Create 10  pods in projects
+    Given I obtain test data file "networking/list_for_pods.json"
+    When I run oc create over "list_for_pods.json" replacing paths:
+      | ["items"][0]["spec"]["replicas"] | 10 |
+    Then the step should succeed
+    Given 10 pod become ready with labels:
+      | name=test-pods |
+
+    #Create an egress ip object
+    Given admin ensures "egressip" egress_ip is deleted after scenario
+    When I obtain test data file "networking/ovn-egressip/egressip1.yaml"
+    And I replace lines in "egressip1.yaml":
+      | 172.31.249.227 | "<%= cb.valid_ips[0] %>" |
+      | qe             | "pm"                     |
+    And I run the :create admin command with:
+      | f | egressip1.yaml |
+    Then the step should succeed
+
+    #Scale down CNO to 0 and delete ovnkube-master pods
+    Given I register clean-up steps:
+    """
+    And I run the :scale admin command with:
+      | resource | deployment                 |
+      | name     | network-operator           |
+      | replicas | 1                          |
+      | n        | openshift-network-operator |
+    Then the step should succeed
+    """
+    # Now scale down CNO pod to 0
+    And I run the :scale admin command with:
+      | resource | deployment                 |
+      | name     | network-operator           |
+      | replicas | 0                          |
+      | n        | openshift-network-operator |
+    Then the step should succeed
+    And admin ensures "ovnkube-master" ds is deleted from the "openshift-ovn-kubernetes" project
+    And admin executes existing pods die with labels:
+      | app=ovnkube-master |
+
+    # Now delete egressip object 
+    Given admin ensures "egressip" egress_ip is deleted
+
+    # Now scale up CNO pod to 1
+    Given I run the :scale admin command with:
+      | resource | deployment                 |
+      | name     | network-operator           |
+      | replicas | 1                          |
+      | n        | openshift-network-operator |
+    Then the step should succeed
+    # A minimum wait for 30 seconds is tested to reflect CNO deployment to be effective which will then re-spawn ovn pods
+    Given 30 seconds have passed
```

Comment:
@huiran0826 hardcoded wait is discouraged.  Isn't the next step atline 824 sufficient?

Reply:
@pruan-rht The next step 824 is not enough under this condition. 
 After  CNO pod scaled to 1, if only step 824, it may pass immediately as all the ovnkube-node-XXX pods are ready status and no ovnkube-master-XXX created yet.  We need to wait ovnkube-master-XXX pods to be recreated.

Reply:
Try with
```
    Given I use the "xxx" project
    And a pod becomes ready with labels:
      | foo=bar |
    Given I run the :scale admin command with:
      | resource | deployment                 |
      | name     | network-operator           |
      | replicas | 1                          |
      | n        | openshift-network-operator |
    Then the step should succeed
    Given I wait for the resource "pod" named "<%= pod.name %>" to disappear
    And admin waits for all pods in the "xxx" project to become ready up to 150 seconds
```

Reply:
We don't need " Given I wait for the resource "pod" named "<%= pod.name %>" to disappear" here,  at that time, still no ovnkube-master-XXX pods, I mean need wait ovnkube-master-XXX pods to be created.
I thought generally we don't use hard code time just avoid to waste waiting time in case it may get ready before the hard code time. Actually in this condition, 30s is not wasted. As you see, next step is still waiting the pods ready.

Reply:
It is to make sure old pod is gone. And next step make sure new pod is up.

We have thousands of test cases. We can not bear test case useless waiting.

Reply:
The old pods was gone check not in this step. It was covered in previous steps. 
Anyway, I will check and try to replace this hard code.

Reply:
Fixed it, thanks!